readme.md

```diff
@@ -0,0 +1,191 @@
+# remark for Visual Studio Code
+
+[![Build][build-badge]][build]
+[![Coverage][coverage-badge]][coverage]
+[![Downloads][downloads-badge]][downloads]
+[![Sponsors][sponsors-badge]][collective]
+[![Backers][backers-badge]][collective]
+[![Chat][chat-badge]][chat]
+
+Visual Studio Code extension to format and lint markdown files with remark.
+
+## Contents
+
+*   [What is this?](#what-is-this)
+*   [When should I use this?](#when-should-i-use-this)
+*   [Use](#use)
+    *   [Formatting](#formatting)
+*   [Plugins](#plugins)
+*   [Syntax](#syntax)
+*   [Compatibility](#compatibility)
+*   [Security](#security)
+*   [Contribute](#contribute)
+*   [License](#license)
+
+## What is this?
+
+This project is a Visual Studio Code (VS Code) extension that you can use in
+your editor to inspect and change markdown files.
+This extension is built around remark, which is a very popular ecosystem of
+plugins that work with markdown.
+You can choose from the 150+ plugins that already exist or make your own.
+
+See [remark][] for info on what the remark ecosystem is.
+
+## When should I use this?
+
+You can use this extension if you want to check (lint) and format markdown
+files from within your editor.
+
+To configure this extension, you define your preferred markdown style in a
+configuration file (`.remarkrc`, `.remarkrc.js`, etc. or in `package.json`).
+This file is picked up by `vscode-remark` and other tools (useful for
+contributors that don’t use VS Code).
+
+The configuration file is also used by [`remark-cli`][remark-cli], which is
+recommended to be used alongside `vscode-remark`, as an npm script and/or in
+CI, to enforce the markdown style.
+
+## Use
+
+To start using this for a project, [`remark-cli`][remark-cli] needs to be
+configured first.
+First, install `remark-cli` and a plugin or preset.
+Let’s use `remark-preset-lint-recommended` for our example.
+
+```sh
+npm install remark-cli remark-preset-lint-recommended
+```
+
+Now create a file named `.remarkrc.json` with the following content:
+
+```json
+{
+  "plugins": ["preset-lint-recommended"]
+}
+```
+
+Now let’s create a markdown file `readme.md` and add the following content:
+
+```markdown
+# My project
+
+This is a [broken link][]
+
+- A `*` should be used for list items
+- Also 4 spaces should be used for indentation
+
+[unused reference]: https://example.com
+```
+
+Now any syntax which doesn’t comply with `remark-preset-lint-recommended` will
+be reported using squiggly lines in your editor and in the `Problems` pane.
+
+### Formatting
+
+This extension can be used as a formatter for markdown files.
+
+To format a file, first press <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>P</kbd>.
```

Comment:
```suggestion
To format a file, first press <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>p</kbd>.
```
```suggestion
To format a file, first press <kbd>Ctrl</kbd> + <kbd>P</kbd>.
```

`shift + p` is implied with `P`?

Reply:
I think it's more user-friendly to be verbose and list all keys to be pressed. Whether the `P` should be lowercase or uppercase is a wish wash; the letter on the physical key is usually an uppercase `P`, but pressing it without shift or caps lock outputs a lowercase `p`. I think I prefer to label what's written on the physical key, but know that's not a water-tight strategy for all symbols across all keyboard layouts.

Reply:
It’s a keyboard shortcut, not an uppercase `P` character somehow combined with the <kbd>Ctrl</kbd> key. Most of the time alphabetic keyboard keys are indicated as upper case, both on the keyboard and in documentation.